mindsdb_native/libs/helpers/text_helpers.py

```diff
@@ -142,37 +142,39 @@ def splitRecursive(word, tokens):
 
 def hashtext(cell):
     text = json.dumps(cell)
-    hash = hashlib.md5(text.encode('utf8')).hexdigest()
-    return hash
+    return hashlib.md5(text.encode('utf8')).hexdigest()
 
 
-def is_foreign_key(data, column_name, data_subtype, other_potential_subtypes):
-    foregin_key_type = DATA_SUBTYPES.INT in other_potential_subtypes or DATA_SUBTYPES.INT == data_subtype
+def _is_foreign_key_name(name):
+    for endings in ['-id', '_id', 'ID', 'Id']:
+        if name.endswith(endings):
+            return True
 
-    data_looks_like_id = True
+    for keyword in ['account', 'uuid', 'identifier', 'user']:
+        if keyword in name:
+            return True
 
-    # Detect UUID
-    if not foregin_key_type:
-        prev_val_length = None
-        for val in data:
-            is_uuid = True
-            is_same_length = True
+    for keyword in ['id']:
+        if keyword == name:
+            return True
+
+    return False
 
-            uuid_charset = set('0123456789abcdef-')
-            set(str(val)).issubset(uuid_charset)
 
-            if prev_val_length is None:
-                prev_val_length = len(str(val))
-            elif len(str(val)) != prev_val_length:
-                is_same_length = False
+def is_foreign_key(data, column_name, data_subtype, other_potential_subtypes):
+    data = list(data)
 
-            prev_val_length = len(str(val))
+    foregin_key_type = DATA_SUBTYPES.INT in other_potential_subtypes or DATA_SUBTYPES.INT == data_subtype
 
-            if not (is_uuid and is_same_length):
-                data_looks_like_id = False
-                break
+    # Detect UUID
+    if foregin_key_type:
+        is_uuid = False
+    else:
+        all_same_length = all(len(str(data[0])) == len(str(x)) for x in data)
+        uuid_charset = set('0123456789abcdef-')
+        all_uuid_charset = all(set(str(x)).issubset(uuid_charset) for x in data)
```

Comment:
If I'm reading this right, `str(x).lower()` would also capture UUID's in all caps.

Reply:
changing `set('0123456789abcdef-')` to `set('0123456789abcdefABCDEF-')` would solve that, right?

Reply:
Yup, that works too!